Document sqlite3mc glibc requirement and SQLite fallback paths in EF Core 11 breaking changes (#5407)
aspnet/EntityFramework.Docs: entity-framework/core/what-is-new/ef-core-11.0/breaking-changes.md

- **Platform (RID) coverage.** SQLite3 Multiple Ciphers doesn't currently include native binaries for every runtime identifier covered by `SourceGear.sqlite3`; for example, `linux-riscv64`, `linux-musl-riscv64`, and `linux-musl-s390x` aren't included. If you target a platform that the new bundle doesn't include, the native library may fail to load at runtime. In that case, revert to the standard build using the package references below.

- **Linux glibc requirement and macOS/Linux opt-out.** The bundled `e_sqlite3mc` library is prebuilt native code. On Linux, it currently requires glibc 2.33 or later; on older distributions, loading it can fail at runtime with an error such as `GLIBC_2.33 not found`. On macOS and Linux, if you need to avoid the bundled library altogether (for example, because of runtime compatibility or because you need to use the system SQLite library), switch to the `.Core` packages and either:
  - Use `SQLitePCLRaw.bundle_e_sqlite3` to keep using the standard `e_sqlite3` bundled build, as shown below.

>   - Use `SQLitePCLRaw.provider.sqlite3` together with a system-installed or self-built SQLite library that matches your deployment environment, and initialize `SQLitePCL.raw.SetProvider(new SQLitePCL.SQLite3Provider_sqlite3());`.

Reviewer: @copilot Refer the reader to the opt-out steps below instead of describing them here as well

- **Platform (RID) coverage.** SQLite3 Multiple Ciphers doesn't currently include native binaries for every runtime identifier covered by `SourceGear.sqlite3`; for example, `linux-riscv64`, `linux-musl-riscv64`, and `linux-musl-s390x` aren't included. If you target a platform that the new bundle doesn't include, the native library may fail to load at runtime. In that case, revert to the standard build using the package references below.

> - **Linux glibc requirement and macOS/Linux opt-out.** The bundled `e_sqlite3mc` library is prebuilt native code. On Linux, it currently requires glibc 2.33 or later; on older distributions, loading it can fail at runtime with an error such as `GLIBC_2.33 not found`. If the target system is unable to satisfy this requirement follow the opt-out steps below.

Reviewer: @copilot don't mention MacOS. Suggest updating glibc

> - **Linux glibc requirement and macOS/Linux opt-out.** The bundled `e_sqlite3mc` library is prebuilt native code. On Linux, it currently requires glibc 2.33 or later; on older distributions, loading it can fail at runtime with an error such as `GLIBC_2.33 not found`. On macOS and Linux, if you need to avoid the bundled library altogether (for example, because of runtime compatibility or because you need to use the system SQLite library), follow the opt-out steps below.

Reviewer: ```suggestion
- **Linux glibc requirement and macOS/Linux opt-out.** The bundled `e_sqlite3mc` library is prebuilt native code. On Linux, it currently requires glibc 2.33 or later; on older distributions, loading it can fail at runtime with an error such as `GLIBC_2.33 not found`. If the target system is unable to satisfy this requirement follow the opt-out steps below.
```

<PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite.Core" Version="11.0.0" />
<PackageReference Include="SQLitePCLRaw.bundle_e_sqlite3" Version="3.x.x" />
```

> If you need to use a system-installed SQLite library instead of a bundled one, reference the provider directly:

Reviewer: ```suggestion
If you need to use a system-installed SQLite library instead of a bundled one, reference `Microsoft.Data.Sqlite.Core` together with `SQLitePCLRaw.provider.sqlite3` instead of the `Microsoft.Data.Sqlite` meta-package:
```
